bson/src/main/org/bson/codecs/pojo/ConventionObjectIdGeneratorsImpl.java

```diff
@@ -30,6 +30,8 @@ public void apply(final ClassModelBuilder<?> classModelBuilder) {
                    classModelBuilder.idGenerator(IdGenerators.OBJECT_ID_GENERATOR);
                } else if (classModelBuilder.getIdGenerator() == null && idType.equals(BsonObjectId.class)) {
                    classModelBuilder.idGenerator(IdGenerators.BSON_OBJECT_ID_GENERATOR);
+               } else if (classModelBuilder.getIdGenerator() == null && idType.equals(String.class)) {
```

Comment:
Wouldn't this only be appropriate if the bson representation of this string property is ObjectId?

Reply:
Yep, I think you're right.

Reply:
Done. I also noticed nothing was really testing this code, so I added a test in `ConvenvtionsTest.java` that tests both `else if` conditions. The first `if` condition is tested in the first test in `ConventionsTest`.